### PR TITLE
Added @param and @returns to docblocks in lib/Extension

### LINKED ISSUE #12223: [Task] Add @param and @return php doc types to /lib/Extension
### Improvement description

- [ ] Add @param php doc types for all public/protected classes/methods/properties in /lib/Extension (excl. Dao and @internal) 
- [ ] Add @return php doc types for all public/protected classes/methods/properties in /lib/Extension (excl. Dao and @internal) 
- [ ] Make sure phpstan tests still run through

### PR DESCRIPTION
**Pull request by aashan10**
## Changes in this pull request  
Resolves #12223

### CODE REVIEW

File: lib/Extension/Bundle/Installer/SettingsStoreAwareInstaller.php [hunk 1]
```diff
@@ -142,18 +142,27 @@ public function uninstall()
...
     }
 
+    /**
+     * @return bool|float|int|string
+     */
     public function isInstalled()
     {
         $installSetting = SettingsStore::get($this->getSettingsStoreInstallationId(), 'pimcore');
 
         return $installSetting ? $installSetting->getData() : false;
```
Comment (fashxp) [inline lib/Extension/Bundle/Installer/SettingsStoreAwareInstaller.php]: we should cast to bool here and just return bool values

**Review by fashxp — CHANGES_REQUESTED** [review]

File: lib/Extension/Bundle/Installer/SettingsStoreAwareInstaller.php [hunk 2]
```diff
@@ -142,18 +142,27 @@ public function uninstall()
...
+    /**
+     * @return bool
+     */
     public function canBeInstalled()
     {
         return !$this->isInstalled();
     }
 
+    /**
+     * @return bool|float|int|string
```
Comment (fashxp) [inline lib/Extension/Bundle/Installer/SettingsStoreAwareInstaller.php]: sorry, missed that one before ... here we also should just allow bool as return value ...

File: lib/Extension/Bundle/Traits/PackageVersionTrait.php [hunk 3]
```diff
@@ -35,7 +35,7 @@ trait PackageVersionTrait
     abstract protected function getComposerPackageName(): string;
 
     /**
-     * {@inheritdoc}
+     * @return array|string|string[]|null
```
Comment (fashxp) [inline lib/Extension/Bundle/Traits/PackageVersionTrait.php]: how can that be an array? Shouldn't that be just a string?

Comment (fashxp) [reply lib/Extension/Bundle/Traits/PackageVersionTrait.php]: I think not even null should be allowed there...

**Review by fashxp — CHANGES_REQUESTED** [review]

File: lib/Extension/Bundle/Traits/PackageVersionTrait.php [hunk 4]
```diff
@@ -49,7 +49,7 @@ public function getVersion()
     }
 
     /**
-     * {@inheritdoc}
+     * @return mixed|string
```
Comment (jdreesen) [inline lib/Extension/Bundle/Traits/PackageVersionTrait.php]: I think this should always be a `string`.

Comment (aashan10) [reply lib/Extension/Bundle/Traits/PackageVersionTrait.php]: Thanks! Changed that

File: lib/Extension/Bundle/Traits/PackageVersionTrait.php (hunk 3 above)
Comment (aashan10) [reply lib/Extension/Bundle/Traits/PackageVersionTrait.php]: I saw that `preg_replace` has those return types that are possible. So, just used those. 
However, I realized that in this scenario, we might just be getting a string. So, changed.

Thanks.

File: lib/Extension/Bundle/Installer/SettingsStoreAwareInstaller.php (hunk 2 above)
Comment (aashan10) [reply lib/Extension/Bundle/Installer/SettingsStoreAwareInstaller.php]: Changed this one as well.

**Review by fashxp — APPROVED** [review]

**Pull request merged**